mixer/adapter/solarwinds/config/config.proto

```diff
@@ -17,33 +17,54 @@ syntax = "proto3";
 package adapter.solarwinds.config;
 
 import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
 
 option go_package = "config";
 option (gogoproto.goproto_getters_all) = false;
 option (gogoproto.equal_all) = false;
 option (gogoproto.gostring_all) = false;
 
 message Params {
-    
+    // AppOptics Access Token needed to send metrics to AppOptics. If no access token is given then metrics
+    // will NOT be shipped to AppOptics
     string appoptics_access_token = 1;
-    
-    string papertrail_url = 2;
 
-    string papertrail_local_retention = 3;
-    
+    // Optional. Max batch size of metrics to be sent to AppOptics.
+    // AppOptics does not allow batch size greater than 1000.
+    // If this is not specified a default batch size of 1000 will be used.
+    int32 appoptics_batch_size = 2;
+
+    // Papertrail url to ship logs to. If no papertrail url is given then the logs will NOT be shipped but rather
+    // dropped.
+    string papertrail_url = 3;
+
+    // This is the duration for which logs will be persisted locally until it is shipped to papertrail in the event
+    // of a network failure
+    google.protobuf.Duration papertrail_local_retention_duration = 4 [(gogoproto.nullable)=true, (gogoproto.stdduration) = true];
+
+    // Describes how to represent an Istio metric in Solarwinds AppOptics
     message MetricInfo {
+        // The name is used to register the metric.
         string name = 1;
-        string instance_name = 2;
-        string description = 3;
-        repeated string label_names = 4;
+
+        // Optional. A human readable description of this metric.
+        string description = 2;
+
+        // The names of labels to use: these need to match the dimensions of the Istio metric.
+        repeated string label_names = 3;
     }
-    repeated MetricInfo metrics = 4;
 
+    // A map of Istio metric name to solarwinds metric info.
+    map<string, MetricInfo> metrics = 5;
+
+    // Describes how to represent an Istio metric in Solarwinds AppOptics
     message LogInfo {
-        repeated string label_names = 1;
-        string payload_template = 2;
-        string instance_name = 3;
+        // Optional. A golang text/template template that will be executed to construct the payload for this log entry.
+        // It will be given the full set of variables for the log to use to construct its result.
+        // If it is not provided, a default template in place will be used.
+        string payload_template = 1;
     }
 
-    repeated LogInfo logs = 5;
+    // A map of Istio logentry name to solarwinds log info.
+    map<string, LogInfo> logs = 6;
```

Comment:
Same thing here, we can validate the key of this map matches one of the instance name for log entries in the validate method.